x-pack/plugins/observability_solution/dataset_quality/server/services/data_telemetry/data_telemetry_service.ts

```diff
@@ -0,0 +1,263 @@
+      );
+
+      return taskInstance;
+    } catch (e) {
+      this.logger?.error(
+        `Failed to schedule task ${LOGS_DATA_TELEMETRY_TASK_ID} with interval ${TELEMETRY_TASK_INTERVAL}. ${e?.message}`
+      );
+
+      return null;
+    }
+  }
+
+  public async isTelemetryOptedIn() {
+    this.isOptedIn = await this.telemetryStart?.getIsOptedIn();
+    return this.isOptedIn === true;
+  }
+
+  private collectAndSend() {
+    // Gather data streams and indices related to each stream of log
+    if (this.esClient) {
+      return getAllIndices({
+        esClient: this.esClient,
+        logsIndexPatterns: LOGS_DATASET_INDEX_PATTERNS,
+        excludeStreamsStartingWith: [...NON_LOG_SIGNALS, ...EXCLUDE_ELASTIC_LOGS],
+        breatheDelay: BREATHE_DELAY_MEDIUM,
+      }).pipe(
+        switchMap((dataStreamsAndIndicesInfo) => {
+          if (dataStreamsAndIndicesInfo.length > MAX_STREAMS_TO_REPORT) {
+            this.logger.debug(
+              `[Logs Data Telemetry] Number of data streams exceeds ${MAX_STREAMS_TO_REPORT}. Skipping telemetry collection.`
+            );
+            return EMPTY;
+          }
+          return of(dataStreamsAndIndicesInfo);
+        }),
+        delay(BREATHE_DELAY_MEDIUM),
+        switchMap((dataStreamsAndIndicesInfo) => {
+          return addMappingsToIndices({
+            esClient: this.esClient!,
+            dataStreamsInfo: dataStreamsAndIndicesInfo,
+            logsIndexPatterns: LOGS_DATASET_INDEX_PATTERNS,
+          });
+        }),
+        delay(BREATHE_DELAY_SHORT),
+        switchMap((dataStreamsAndIndicesInfo) => {
+          return addNamespace({
+            dataStreamsInfo: dataStreamsAndIndicesInfo,
+          });
+        }),
+        delay(BREATHE_DELAY_MEDIUM),
+        switchMap((infoWithNamespace) => {
+          return getIndexBasicStats({
+            esClient: this.esClient!,
+            indices: infoWithNamespace,
+            breatheDelay: BREATHE_DELAY_MEDIUM,
+          });
+        }),
+        delay(BREATHE_DELAY_SHORT),
+        switchMap((infoWithStats) => {
+          return getIndexFieldStats({
+            basicStats: infoWithStats,
+          });
+        }),
+        delay(BREATHE_DELAY_SHORT),
+        map((statsWithNamespace) => {
+          return groupStatsByPatternName(statsWithNamespace);
+        }),
+        map((statsByPattern) => {
+          return indexStatsToTelemetryEvents(statsByPattern);
+        }),
+        delay(BREATHE_DELAY_SHORT),
+        switchMap((dataTelemetryEvents) => {
+          return from(this.reportEvents(dataTelemetryEvents));
+        })
+      );
+    } else {
+      this.logger.warn(
+        `[Logs Data Telemetry] Elasticsearch client is unavailable: cannot retrieve data streams
+        for stream of logs`
+      );
+
+      return EMPTY;
+    }
+  }
+
+  private async reportEvents(events: DataTelemetryEvent[]) {
+    // TODO: Implement reporting events via analytics service
```

Comment:
> [!NOTE]
> Is this something that should be addressed on this PR or in future? Is there a need to keep it here at this stage?

Reply:
It'll be implemented in the subsequent PR. In this PR, it's to define the structure and to assert in tests if the function is called at the appropriate time.

Reply:
Just a minor suggestion, as seen in other task-manager's tasks pre-aggregating data like this one:

The other examples return the pre-aggregated data (the events here) in the task runner. This makes taskManager store them in the task's last-execution status.

Then, they register a usage collection `usageCollection.makeUsageCollector` that reads the latest state of their task, and returns that payload.

Just an idea of how to ship it 😇

Reply:
@afharo it's a good suggestion. In fact, it's a question that we are looking into that whether can we use the existing usage collection channel for this telemetry. As already many plugins push telemetry via usage collector we wonder if it's gonna cause delays/misses if we use the same channel, instead of a separate EBT channel. The nature of data is of snapshot telemetry though. What are your thoughts on this?